cv32e40s/tests/programs/custom/xsecure_test/xsecure_test.c

```diff
@@ -0,0 +1,185 @@
+#ifndef __XSECURE_TEST_H__
+#define __XSECURE_TEST_H__
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+//TODO: functions are fetched from the privilege test code. If these functions end up in a common function file, we should adopt this file to use the common functions.
+
+#define CPUADDR_CPUCTRL     0xBF0
+#define CPUADDR_SECURESEED0 0xBF9
+#define CPUADDR_SECURESEED1 0xBFA
+#define CPUADDR_SECURESEED2 0xBFC
+
+// macros for using defines in inline asm
+#define S(x) #x
+#define STR(s) S(s)
+
+// global values to hold registers for tests
+volatile uint32_t cpuctrl, secureseed0, secureseed1, secureseed2;
+
+int read_ctrlcpu_and_secureseeds_in_user_mode(void);
+int read_ctrlcpu_and_secureseeds_in_machine_mode(void);
+
+
+// assembly function to setup a generous PMP-region for user mode.
+extern volatile void  setup_pmp();
+// assembly function to set privilege-mode to user-mode
+extern volatile void set_u_mode();
+
+// standard value for the mstatus register
+#define MSTATUS_STD_VAL 0x1800
+
+typedef enum {
+  M_MODE_BEH,
+  TRAP_INCR_BEH
+} trap_behavior_t;
+
+// trap handler behavior definitions
+volatile trap_behavior_t trap_handler_beh;
+
+volatile uint32_t NUM_TRAP_EXECUTIONS;
+
+
+/* Declaration of assert */
+static void assert_or_die(uint32_t actual, uint32_t expect, char *msg) {
+  if (actual != expect) {
+    printf(msg);
+    printf("expected = 0x%lx (%ld), got = 0x%lx (%ld)\n", expect, (int32_t)expect, actual, (int32_t)actual);
+    exit(EXIT_FAILURE);
+  }
+}
+
+/* Retuns specific bit-field from [bit_indx_low : bit_indx_high] in register x */
+unsigned int get_field(unsigned int x, int bit_indx_low, int bit_indx_high){
+    int field = ( 1 << ( (bit_indx_high - bit_indx_low) + 1) )  - 1;
```

Comment:
`field` is acting as a `mask`, isn't it?
Since this function gets reused a lot I think it could be made clearer to read.

`(bit_indx_high - bit_indx_low) + 1` is acting as `field_length`?

`x` (when used as the output) is actually the real `field`?

Reply:
Actually the function isnt used at all. So ill just remove it